main.tf

```diff
@@ -280,6 +280,23 @@ resource "aws_codebuild_project" "default" {
     }
   }
 
+  dynamic "secondary_sources"{
+      iterator = secondary_source
+      for_each = length(var.secondary_sources) > 0 ? var.secondary_sources : []
+      content {
+        git_clone_depth = secondary_source.value.git_clone_depth
```

Comment:
need to add fallback to `null` for non-required vars (see https://github.com/cloudposse/terraform-aws-codebuild/blob/974c9d579ca18e4a19a11c9ae1dd858ee76de7b9/main.tf#L324 as example)